src/arithmetic/curves.rs

```diff
@@ -131,3 +133,49 @@ pub trait CurveAffine:
     /// Returns the curve constant $b$.
     fn b() -> Self::Base;
 }
+
+/// The affine coordinates of a point on an elliptic curve.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Coordinates<C: CurveAffine> {
+    pub(crate) x: C::Base,
+    pub(crate) y: C::Base,
+}
+
+impl<C: CurveAffine> Coordinates<C> {
+    /// Returns the x-coordinate.
+    ///
+    /// Equivalent to `Coordinates::u`.
+    pub fn x(&self) -> &C::Base {
+        &self.x
+    }
```

Comment:
AFAIK this doesn't allow going from CtOption<Coordinates<C>> to CtOption<C>, which seems useful.

Reply:
Yes it does, and I use it for exactly this purpose.

Reply:
Ah, I misread. We don't need to go to `CtOption<C>` for Extract, but we _can_ just use `.map(|c| CurveAffine::from_xy(c.x(), c.y()))`